# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MCallOptimize.cpp
# Commit: 7805ddfe2bfa
# Full Hash: 7805ddfe2bfa83383dc422ed1e8ac70bda627a89
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Rename TypeCode/ValType/RefType::Any to Extern. r=lth
#   
#   This commit renames the TypeCode/ValType/RefType occurrences of 'any' to 'extern'
#   to match what they actually represent. All public API facing occurrences have
#   already been switched to 'externref' so this should only be aligning the
# ==============================================================================

diff -r 4fa52a0617de -r 7805ddfe2bfa js/src/jit/MCallOptimize.cpp
--- a/js/src/jit/MCallOptimize.cpp	Tue Aug 18 16:58:21 2020 +0000
+++ b/js/src/jit/MCallOptimize.cpp	Tue Aug 18 16:58:32 2020 +0000
@@ -4042,7 +4042,7 @@
         MOZ_CRASH("impossible per above check");
       case wasm::ValType::Ref:
         switch (sig.args()[i].refTypeKind()) {
-          case wasm::RefType::Any:
+          case wasm::RefType::Extern:
             // Transform the JS representation into an AnyRef representation.
             // The resulting type is MIRType::RefOrNull.  These cases are all
             // effect-free.